internal/artifacts/module_arch.go

```diff
@@ -108,117 +115,60 @@ func getBuildErrMsg(oneModuleMsg, manyModulesMsg string, modules []string) strin
 	return manyModulesMsg
 }
 
-func buildSelectedModules(source, target string, extensions []string, mtaObj *mta.MTA, selectedModules []string,
-	allDependencies bool, wdGetter func() (string, error)) error {
-
-	processedModules := make(map[string]bool)
+func collectSelectedModulesAndDependencies(mtaObj *mta.MTA, modulesWithDependencies map[string]bool, moduleName string) error {
 
-	for _, module := range selectedModules {
-		err := buildSelectedModule(source, target, extensions, mtaObj, module, allDependencies, processedModules, wdGetter)
+	modulesWithDependencies[moduleName] = true
+	module, err := mtaObj.GetModuleByName(moduleName)
+	if err != nil {
+		return err
+	}
+	for _, requires := range buildops.GetBuildRequires(module) {
+		requiredModule, err := mtaObj.GetModuleByName(requires.Name)
+		if err != nil {
+			return err
+		}
 
+		err = collectSelectedModulesAndDependencies(mtaObj, modulesWithDependencies, requiredModule.Name)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func buildSelectedModule(source, target string, extensions []string, mtaObj *mta.MTA, module string,
-	allDependencies bool, processedModules map[string]bool, wdGetter func() (string, error)) error {
+func buildModules(source, target string, extensions []string, selectedModules []string,
+	selectedModulesMap map[string]bool, wdGetter func() (string, error)) error {
 
-	logs.Logger.Infof(buildMsg, module)
+	for _, module := range selectedModules {
+		err := buildSelectedModule(source, target, extensions, module, selectedModulesMap, wdGetter)
 
-	if allDependencies {
-		err := processModuleDependencies(source, target, extensions, mtaObj, module, processedModules, wdGetter)
 		if err != nil {
 			return err
 		}
 	}
+	return nil
+}
+
+func buildSelectedModule(source, target string, extensions []string, module string,
+	selectedModulesMap map[string]bool, wdGetter func() (string, error)) error {
+
+	logs.Logger.Infof(buildMsg, module)
 
 	moduleLoc, err := getModuleLocation(source, target, module, extensions, wdGetter)
 	if err != nil {
 		return err
 	}
 
-	err = buildModule(moduleLoc, moduleLoc, module, "", false, true)
+	err = buildModule(moduleLoc, moduleLoc, module, "", false, selectedModulesMap[module])
```

Comment:
I'm a bit bothered that we parse the mta.yaml for every module. It would be better to maybe initialize the location object with the parsed mta object (or maybe cache it on the first parse). But since it's the same flow as in the regular build it shouldn't be done in this PR.